projects/ngx-maplibre-gl/src/lib/map/map.service.spec.ts

```diff
@@ -208,13 +210,90 @@ describe('MapService', () => {
       popupOpen: { emit: jasmine.createSpy() },
       popupClose: { emit: jasmine.createSpy() },
     } as any;
-    const popup = mapService.createPopup({ 
-      popupOptions: {}, 
+    const popup = mapService.createPopup({
+      popupOptions: {},
       popupEvents
     }, container);
     mapService.addPopupToMap(popup, [0, 0]);
     mapService.removePopupFromMap(popup);
     popup.fire('close');
     expect(popupEvents.popupClose.emit).not.toHaveBeenCalled();
   });
+
+  describe('layer handling', () => {
+    it('should unsubscribe from events on removeLayer via source', async () => {
+      spyOn(mapService.mapInstance, 'queryRenderedFeatures').and.returnValue([{} as MapGeoJSONFeature]);
+
+      const sourceData: SourceSpecification = {
+        "type": "geojson",
+        "data": {
+          "type": "FeatureCollection",
+          "features": []
+        }
+      }
+      const layerEventThatShouldNotEmit = {
+        ...createLayerEvents(),
+        layerClick: { emit: jasmine.createSpy() } as any,
+      };
+      const layer: SetupLayer = {
+        layerOptions: {
+          type: 'fill',
+          id: 'layerId',
+          source: "sourceId",
+          paint: {},
+        },
+        layerEvents: layerEventThatShouldNotEmit,
+      };
+      const layer2: SetupLayer = {
+        layerOptions: {
+          type: 'fill',
+          id: 'layerId',
+          source: "sourceId",
+          paint: {},
+        },
+        layerEvents: createLayerEvents(),
+      };
+      const mapLoaded = new ReplaySubject(1);
+      mapEvents.mapLoad.subscribe(() => {
+        mapLoaded.next(true);
+      });
+      await firstValueFrom(
+        mapLoaded.pipe(
+          tap(() => {
```

Comment:
Is all this needed?
Can't you simply `await firstValueFrom(mapEvents.mapLoad);` and then check all the asserts?